rudaux/submission.py

```diff
@@ -98,12 +98,26 @@ def initialize_submission_sets(config, course_infos, assignments, students, subm
                                         'student' : stu,
                                         'name' : f"{course_name}-{course_info['id']} : {assignment['name']}-{assignment['id']} : {stu['name']}-{stu['id']}"
                                         } for stu in students[i] if stu['status'] == 'active']
+
+            subms_to_remove=[]
+
             for subm in subm_set[course_name]['submissions']:
                 student = subm['student']
+
+                # Add check for SD student, save for later removal.
+                if student['id'] not in subm_info[assignment['id']]:
+                    subms_to_remove.append(subm)
+                    continue
```

Comment:
This is an interesting change. I hadn't considered SD students -- but basically in the past rudaux just treated them as regular students and gave them 0 for missing work (but we ignore those grades at the end of the semester anyway so it doesnt matter)

So I'm not sure this change is needed...and I do worry a bit that 

```
if student['id'] not in subm_info[assignment['id']]:
```
might accidentally remove a regular student. What is in `subm_info`? I forget

Reply:
(If you're certain this won't remove any regular students i'm happy w it tho)

Reply:
I think sumb_info is a list of dictionaries, one for each assignment with one key assignment id whose value is another dict with submissions for each student by student id. 

The issue was that the SD students weren't assigned anything which caused a KeyError with student id when trying to access subm['score'] = subm_info[assignment['id']][student['id']].